S3FileRenamer.py

```diff
@@ -6,35 +6,48 @@
     Default behavior is to only generate the log file for review.  Pass the "--rename" or "-r" flag to rename files.
  """
 
-from os import walk, rename
-from os.path import join
+import os
 import argparse
 import re
 
-
-def contains_bad_characters(string):
+def contains_bad_characters(string, badchars):
     """Search input string for illegal characters.  Return True, list of characters found, if any found."""
     found_bad_characters = []
     for char in string:
-        if char in bad_characters:
+        if char in badchars:
             found_bad_characters.append(char)
     if len(found_bad_characters) > 0:
         return True, found_bad_characters
     else:
         return False, ""
 
-
 def perform_rename(old_name, root_path, found_bad_characters, rename_do):
     """Remove illegal characters if rename argument was passed to program."""
-    corrected_name = re.sub("|".join(found_bad_characters), "", old_name)
+# Replaced "re.sub" because of conflict with regular expression wildcards
+#   corrected_name = re.sub("|".join(found_bad_characters), "", old_name)
+    corrected_name = ""
+    for c in old_name:
+        if c in found_bad_characters:
+            c = "_"                     # Replacement character can be ""
+        corrected_name += c
+    # Avoid overwriting files that already exist.
+    n = 0
+    done = False
+    check_name = corrected_name
+    while not done:
+        if os.path.exists(os.path.join(root_path, check_name)):
```

Comment:
You could just change ```from os.path import join``` to ```from os.path import join, exists``` and change ```os.path.exists(......)``` to ```exists(........)```

Then you don't need to do any other changes(line 45-46,60,64,69) due to your change in line 9

Reply:
I agree with @yinlinchen.  That increases the verbosity without improving readability.  Please revert to importing individual functions rather than entire modules.